src/Language/Haskell/Liquid/Bare/Resolve.hs

```diff
@@ -535,11 +537,24 @@ instance ResolveSym Ghc.DataCon where
                     Ghc.AConLike (Ghc.RealDataCon x) -> Just x 
                     _                                -> Nothing
 
-instance ResolveSym F.Symbol where 
-  resolveLocSym env name _ lx = case resolveLocSym env name "Var" lx of 
-    Left _               -> Right (val lx)
+instance ResolveSym F.Symbol where
+  resolveLocSym env name _ lx = case resolveLocSym env name "Var" lx of
+    -- If we can't resolve the input 'Symbol' from an 'Id', try again
+    -- by grabbing the 'Id' of an 'AConLike', if any.
+    -- See test T1688.
+    Left _               -> case resolveLocSym env name "Var" lx of
+      Left _  -> Right (val lx)
+      Right v -> Right (F.symbol $ unWrap (v :: Wrapped Ghc.Var))
     Right (v :: Ghc.Var) -> Right (F.symbol v)
 
+newtype Wrapped a = Wrapped { unWrap :: a } deriving (PPrint)
+
+instance ResolveSym (Wrapped Ghc.Var) where
+  resolveLocSym = resolveWith "variable" $ \case
+                    Ghc.AConLike (Ghc.RealDataCon x) -> Just $ Wrapped (Ghc.dataConWorkId x)
+                    _                                -> Nothing
```

Comment:
Again, could we document what the difference in practice is between this case and the `DataCon` case?

Independently, I don't see the value of introducing a dummy type constructor (`Wrapped`) just so that we can make this an instance, which then presumably is called in just a single place in the program, namely in line 547 above? So why not inline this code into the `Symbol` instance?